src/modules/core/components/Popover/Popover.tsx

```diff
@@ -116,30 +120,37 @@ const Popover = ({
 
   const lastIsOpenProp = usePrevious(isOpenProp);
 
+  const close = useCallback(
+    (data?: any, modifiers?: { cancelled: boolean }) => {
+      if (openTimeoutRef.current) {
+        clearTimeout(openTimeoutRef.current);
+      }
+      if (closeTimeoutRef.current) {
+        clearTimeout(closeTimeoutRef.current);
+      }
+      _setIsOpen(false);
+      if (typeof onClose == 'function') onClose(data, modifiers);
+    },
+    [onClose],
+  );
+
   const requestOpen = useCallback(() => {
     if (isOpen) {
       return;
     }
-    if (openDelay && window) {
-      openTimeoutRef.current = window.setTimeout(() => {
+    if (closeAfterDelay) {
+      closeTimeoutRef.current = setTimeout(() => {
+        close();
+      }, 3000 + (openDelay || 0));
+    }
+    if (openDelay) {
+      openTimeoutRef.current = setTimeout(() => {
         _setIsOpen(true);
       }, openDelay);
       return;
     }
     _setIsOpen(true);
-  }, [isOpen, openDelay]);
-
-  const close = useCallback(
-    (data?: any, modifiers?: { cancelled: boolean }) => {
-      if (window) {
-        window.clearTimeout(openTimeoutRef.current);
-      }
-      if (!isOpen) return;
-      _setIsOpen(false);
-      if (typeof onClose == 'function') onClose(data, modifiers);
-    },
-    [isOpen, onClose],
-  );
+  }, [isOpen, openDelay, closeAfterDelay]);
```

Comment:
> What's the state of this? I've been testing with metamask and the gas station is still closing itself when the transaction is in progress

@ArmandoGraterol I got stuck on figuring out how to determine when all transactions have been completed successfully when connected to Metamask, so I started working on the Gnosis and the Decisions tasks which both seemed to be higher priority features 😅. I can resume work as soon as I am done with the Add Decision Preview Page task. However, if you have any suggestions for how to make this work, please let me know.

Reply:
@Julianahlk I took a look into this to see if I could offer a solution, and I've made a commit to this branch with what I found.  It checks if all the transactions in the most recent group have succeeded, and if so, will close the popover after the delay. Let me know your thoughts 🙂

Reply:
@ArmandoGraterol I made some more updates to get this working so that the popover only closes when the transactions have succeeded but it works in most cases and sometimes it closes before all transactions have succeeded. Could you take a look?, I could use some help figuring out what the issue is. Thanks.

Reply:
@willm30 I just noticed your comment from earlier now, so weird that I didn't see it 😅 Anyway, I also didn't see your commit here in Github. Are you sure you pushed the commit?.